edelphi/src/main/java/fi/metatavu/edelphi/query/thesis/Multiple2DScalesThesisQueryPageHandler.java

```diff
@@ -0,0 +1,201 @@
+
+        if (commentable) {
+          QueryQuestionComment comment = queryQuestionCommentDAO.findRootCommentByQueryReplyAndQueryPage(queryReply, queryPage);
+          exportContext.addCellValue(queryReply, commentColumnIndex, comment != null ? comment.getComment() : null);
+        }
+      }
+    }
+  }
+
+  @Override
+  public List<QueryOption> getDefinedOptions() {
+    List<QueryOption> result = new ArrayList<>(super.getDefinedOptions());
+    result.addAll(this.options);
+    return result;
+  }
+  
+  @Override
+  public void updatePageOptions(Map<String, String> settings, QueryPage queryPage, User modifier, boolean hasAnswers) {
+    super.updatePageOptions(settings, queryPage, modifier, hasAnswers);
+
+    for (QueryOption queryOption : getDefinedOptions()) {
+      if ((queryOption.getType() == QueryOptionType.QUESTION) && ((!hasAnswers) || (queryOption.isEditableWithAnswers()))) {
+        QueryPageUtils.setSetting(queryPage, queryOption.getName(), settings.get(queryOption.getName()), modifier);
+      }
+    }
+    
+    String thesesOption = settings.get(THESES_OPTION);
+    String labelX = settings.get(LABEL_X_OPTION);
+    String labelY = settings.get(LABEL_Y_OPTION);
+    List<String> fieldOptions = QueryPageUtils.parseSerializedList(settings.get(OPTIONS_OPTION));
+    List<String> theses = QueryPageUtils.parseSerializedList(thesesOption);
+
+    for (int thesisIndex = 0, thesisCount = theses.size(); thesisIndex < thesisCount; thesisIndex++) {
+      String thesis = theses.get(thesisIndex);
+      
+      String fieldNameX = getFieldName(thesisIndex, "x");
+      String fieldNameY = getFieldName(thesisIndex, "y");
+      String fieldLabelX = getFieldLabel(thesis, labelX);
+      String fieldLabelY = getFieldLabel(thesis, labelY);
+      Boolean mandatory = false;
+      
+      if (hasAnswers) {
+        synchronizeFieldCaption(queryPage, fieldNameX, fieldLabelX);
+        synchronizeFieldCaption(queryPage, fieldNameY, fieldLabelY);
+      } else {
+        synchronizeField(queryPage, fieldOptions, fieldNameX, fieldLabelX, mandatory);
+        synchronizeField(queryPage, fieldOptions, fieldNameY, fieldLabelY, mandatory);
+      }
+    }
+  }
+  
+  private void addFragmentStringList(RequiredQueryFragment requiredFragment, String name, List<String> list) {
+    requiredFragment.addAttribute(String.format("%sCount", name), String.valueOf(list.size()));
+    for (int i = 0, l = list.size(); i < l; i++) {
+      requiredFragment.addAttribute(String.format("%s.%d", name, i), list.get(i));
+    }
+  }
+
+  private String getFieldName(int index, String axis) {
+    return String.format("multiple2dscales.%d.%s", index, axis);
+  }
+  
+  private String getFieldLabel(String thesis, String label) {
+    return String.format("%s/%s", thesis, label);
+  }
+  
+  private String getColumnLabel(String pageTitle, String fieldCaption) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getColumnLabel" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)